### ISSUE TITLE
Route PUT /source/<project>/<package>/<filename> has the wrong schema documented

### ISSUE THREAD

**Issue by dcermak** [openSUSE/open-build-service#9706]
# Issue Description

The route `PUT /source/<project>/<package>/<filename>` is returning a `result.xsd` according to the  [documentation](https://build.opensuse.org/apidocs/index#67). This is however not true when invoking it directly via `curl --user ${CREDENTIALS} -X PUT ${OBS}/source/${project}/${package}/${filename} -d ""`, which returns a revision.

Expected Result
============
Either the route should return a status or the documentation should state that it returns a revision.

How to Reproduce
===============
1. Create a dummy package
2. Run: 
```ShellSession
$ curl --user ${CREDENTIALS} -X PUT ${OBS}/source/${project}/${package}/${filename} -d ""
<revision rev="1" vrev="1">
  <srcmd5>12f1ca68b9ac8cbd5767771391aaa445</srcmd5>
  <version>unknown</version>
  <time>1591707122</time>
  <user>Admin</user>
  <comment></comment>
  <requestid/>
</revision>
```

Further Information
===============
* Tested on OBS 2.10.5

**Comment by dcermak** [comment]
Yet another inconsistency: if you instead use `curl --user ${CREDENTIALS} -X PUT ${OBS}/source/${project}/${package}/${filename}?rev=repository -d "something in here"` then you will get neither a status nor a revision back, but this:
```xml
<revision rev="repository">
  <srcmd5>d41d8cd98f00b204e9800998ecf8427e</srcmd5>
</revision>
```
This does look like a revision, but does not adhere to the schema as defined in [`revisionlist.rng`](https://build.opensuse.org/apidocs/revisionlist.rng) as the elements `version`, `time` and `user` are not optional.

**Comment by mlschroe** [comment]
Repository only uploads (`rev=repository`) do not create a revision at all, so you only get the srcmd5 back.

**Comment by dcermak** [comment]
> Repository only uploads (`rev=repository`) do not create a revision at all, so you only get the srcmd5 back.

That makes sense, but of which object do I get the md5 hash? I've tried to figure it out on a local OBS test instance, but have failed so far.

(And the documentation should still explain that imho)

**Comment by mlschroe** [comment]
I don't understand that. You can use the returned `srcmd5` hash as revision if you want to access your files. (i.e. `rev=d41d8cd98f00b204e9800998ecf8427e` in your example)

**Comment by dcermak** [comment]
Let me rephrase that: is the returned `srcmd5` the hash of the file that I just uploaded or the package in the new "dirty" state or of the package in the state without the modifications or something entirely different?

**Comment by mlschroe** [comment]
The returned srcmd5 is a representation of the complete file set. Git calls those "tree" objects.

**Comment by dcermak** [comment]
Michael Schroeder <notifications@github.com> writes:

> The returned srcmd5 is a representation of the complete file set. Git calls those "tree" objects.

Including the changes that were just pushed? So essentially if do a

curl -X PUT /source/${project}/${package}/${file}?rev=repository -d "$contents"

followed by a commit, then the new revision hash of the commit will
equal to the srcmd5 value that I received from the PUT?

**Comment by marcus-h** [comment]
On 2020-06-09 09:48:27 -0700, Dan Čermák wrote:
> Michael Schroeder <notifications@github.com> writes:
>
> > The returned srcmd5 is a representation of the complete file set. Git calls those "tree" objects.
>
> Including the changes that were just pushed? So essentially if do a
>
> curl -X PUT /source/${project}/${package}/${file}?rev=repository -d "$contents"
>
> followed by a commit, then the new revision hash of the commit will
> equal to the srcmd5 value that I received from the PUT?
>
No. The PUT (with rev=repository) always returns the srcmd5 of the empty
file set (it is always the same and not related to the file you PUTed (it
is just the output of "echo -n | md5sum")).
The srcmd5 returned from a commit identifies the file set that you just
committed.

**Comment by dcermak** [comment]
Marcus Hüwe <notifications@github.com> writes:

> On 2020-06-09 09:48:27 -0700, Dan Čermák wrote:
>> Michael Schroeder <notifications@github.com> writes:
>>
>> > The returned srcmd5 is a representation of the complete file set. Git calls those "tree" objects.
>>
>> Including the changes that were just pushed? So essentially if do a
>>
>> curl -X PUT /source/${project}/${package}/${file}?rev=repository -d "$contents"
>>
>> followed by a commit, then the new revision hash of the commit will
>> equal to the srcmd5 value that I received from the PUT?
>>
> No. The PUT (with rev=repository) always returns the srcmd5 of the empty
> file set (it is always the same and not related to the file you PUTed (it
> is just the output of "echo -n | md5sum")).

So that means that the response is *always* the same and can be very
well ignored? Why is it even send then?
